python/fledge/plugins/south/person_detection/person_detection.py

```diff
@@ -210,12 +240,32 @@ def plugin_reconfigure(handle, new_config):
         new_handle: new handle to be used in the future calls
     Raises:
     """
+    global frame_processor
+
+    parameters_to_check = ['camera_id', 'enable_window', 'enable_web_streaming', 'web_streaming_port_no']
+    need_to_shutdown = check_need_to_shutdown(handle, new_config, parameters_to_check)
+    if not need_to_shutdown:
+        _LOGGER.debug("No need to shutdown")
+        frame_processor.handle_new_config(new_config)
+        new_handle = plugin_init(new_config)
+        return new_handle
+    
+    else:
 
-    plugin_shutdown(handle)
-    new_handle = plugin_init(new_config)
-    plugin_start(new_handle)
+        was_native_window_enabled = handle['enable_window']['value'] == 'true'
+        is_native_window_enabled_now = new_config['enable_window']['value'] == 'true'
 
-    return new_handle
+        if was_native_window_enabled or is_native_window_enabled_now:
```

Comment:
"need_to_shutdown" is false here, so was_ & is_ variables would have same value. So it is not really "enabling the native window during reconf". The native window would remain unchanged. May be explain the overall logic in this function?

Reply:
The motive is not let changing the enable / disable window at all during reconfigure.  

 There could be four cases 
1. Previous disabled Now enabled 
2. Previous Enabled Now Enabled 
3. Previous Enabled Now disabled
4. Previous Disabled Now Disabled

Only the fourth case should be allowed. case 1 is risky for sure. Case 2 and 3 are also some what risky. Better consider them as 1 only.

Reply:
Done